src/goto-instrument/contracts/contracts.h

```diff
@@ -123,7 +123,7 @@ class code_contractst
   void check_apply_loop_contracts(
     goto_functionst::goto_functiont &goto_function,
     const local_may_aliast &local_may_alias,
-    const goto_programt::targett loop_head,
+    goto_programt::targett loop_head,
```

Comment:
I didn't see any modification in `loop_head`, is this change necessary?

Reply:
Yes, you can look at line 229 and the `insert_before_swap_and_advance` function (specifically, line 90 in that function).

We insert some code (havoc & assumptions) before the loop head, and this change should have been made earlier. We didn't care about it because we were not really using the `loop_head` after that insertion. Now that we do, we need to advance if correctly.

Reply:
Thanks for the clarification!